lib/components/Buttons/ButtonSecondary.tsx

```diff
@@ -1,65 +0,0 @@
-import React, { FC, ReactNode } from "react";
-import * as styles from "./styles";
-import { Link } from "../types";
-import { cx } from "@emotion/css";
-
-interface Props {
-  label: ReactNode;
-  className?: string;
-  onClick?: () => void;
-  href?: string;
-  variant?: "gray" | "icon" | "blueRounded" | null;
-  link?: Link;
-  rel?: string;
-  target?: string;
-  isExternalHref?: boolean;
-  disabled?: boolean;
-}
-
-export const ButtonSecondary: FC<Props> = ({ link: LinkElement, ...props }) => {
-  let buttonStyle;
-  if (props.variant === "gray") {
-    buttonStyle = cx(
-      styles.button_secondary,
-      styles.button_secondary_gray,
-      props.className
-    );
-  } else if (props.variant === "blueRounded") {
-    buttonStyle = cx(
-      styles.button_secondary,
-      styles.button_secondary_blue,
-      props.className
-    );
-  } else {
-    buttonStyle = cx(styles.button_secondary, props.className, {
-      icon: props.variant === "icon",
-    });
-  }
-
-  if (props.href) {
-    if (LinkElement) {
-      return (
-        <LinkElement href={props.href}>
-          <a className={buttonStyle}>{props.label}</a>
-        </LinkElement>
-      );
-    }
-
-    return (
-      <a {...props} className={buttonStyle} href={props.href}>
-        {props.label}
-      </a>
-    );
-  }
-
-  return (
-    <button
-      type="button"
-      className={buttonStyle}
-      onClick={props.onClick}
-      disabled={props.disabled}
-    >
-      {props.label}
-    </button>
-  );
-};
```

Comment:
I would keep the component `ButtonSecondary` 

and create a BaseButton Component for both `ButtonPrimary` and `ButtonSecondary` which pass their styles to this component.


So like


```TS
// BaseButton
export interface Props {
  label: ReactNode;
  className?: string;
  onClick?: () => void;
  href?: string;
  variant?: "gray" | "icon" | "blueRounded" | null;
  link?: Link;
  rel?: string;
  target?: string;
  isExternalHref?: boolean;
  disabled?: boolean;
}

interface BaseProps extends Props {
  buttonStyle: string;
}

export const BaseButton: FC<BaseProps> = ({ link: LinkElement, ...props }) => {
  if (props.href) {
    if (LinkElement) {
      return (
        <LinkElement href={props.href}>
          <a className={props.buttonStyle}>{props.label}</a>
        </LinkElement>
      );
    }

    return (
      <a {...props} className={props.buttonStyle} href={props.href}>
        {props.label}
      </a>
    );
  }

  return (
    <button
      type="button"
      className={props.buttonStyle}
      onClick={props.onClick}
      disabled={props.disabled}
    >
      {props.label}
    </button>
  );
};
```

Reply:
Button primary

```TS

import { BaseButton, Props } from "./BaseButton";


export const ButtonPrimary: FC<Props> = (props) => {
  const buttonStyle = cx(
    styles.buttonPrimary,
    {
      gray: props.variant === "gray",
    },
    {
      blueRounded: props.variant === "blueRounded",
    },
    {
      icon: props.variant === "icon",
    },
    props.className
  );

  return <BaseButton {...props} buttonStyle={buttonStyle} />;
};
```

Reply:
Button Secondary

```TS

import { BaseButton, Props } from "./BaseButton";

export const ButtonSecondary: FC<Props> = (props) => {
  const buttonStyle = cx(
    styles.buttonSecondary,
    {
      gray: props.variant === "gray",
    },
    {
      blueRounded: props.variant === "blueRounded",
    },
    {
      icon: props.variant === "icon",
    },
    props.className
  );
  return <BaseButton {...props} buttonStyle={buttonStyle} />;
};
```